### PR TITLE
 [StructuredOutput] Changes expander icon to match with TreeView expander behaviour

### PR DESCRIPTION
**Pull request by netonjm**
expander

This task introduces a new challenge in how we are drawing the elements in our row, because it adds an expander in the first view position and depends in the row text fits or not the text column.

To fix this we have needed a better/dynamic way to stack all the row views. Right now we do all this frames calculations in CalculateLayout and Render method works with this data passed as out parameters (the same in Render/click/mouse move). This out parameters generates many unnecessary allocations and duplicates same data we have in our ViewState.

With this change we introduced new frame cache for row Icon, and changed the way of how we calculate the stack of views depending if fits or not. 

With this changes move to first element the expander cost a few lines of code:
https://github.com/mono/monodevelop/pull/5461/commits/9296dcb7558390ac0d2dd5aa00f8c733aa2c48b5 

**We need ensure we don't have any regressions or something new is broken**

Fixes Bug #595915

https://devdiv.visualstudio.com/DevDiv/_workitems/edit/595915

Design task is here: 
https://devdiv-design.visualstudio.com/DevDiv%20Design/_workitems/edit/2883?src=WorkItemMention&src-action=artifact_link

### CODE REVIEW

**Comment by netonjm** [comment]
Now the commits are more clear.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.csproj [hunk 1]
```diff
@@ -3037,6 +3037,30 @@
     <EmbeddedResource Include="icons\build-task-success-16~dark~sel%402x.png" />
     <EmbeddedResource Include="icons\build-task-success-16~sel.png" />
     <EmbeddedResource Include="icons\build-task-success-16~sel%402x.png" />
+    <EmbeddedResource Include="icons\tree-chevron-closed.png">
```
[the remark this answers is not in the document]
Comment (netonjm) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.csproj]: The resources are taken from this design task: https://devdiv-design.visualstudio.com/DevDiv%20Design/_workitems/edit/2883?src=WorkItemMention&src-action=artifact_link

**Comment by sgmunn** [comment]
@monojenkins back port release-7.6

**Comment by sgmunn** [comment]
@monojenkins backport release-7.7

**Comment by sgmunn** [comment]
@monojenkins backport release-7.6

**Comment by sgmunn** [comment]
@rodrmoya / @netonjm please back port to release-7.6 tomorrow am for inclusion in RTW.

**Comment by rodrmoya** [comment]
Manual backport done at https://github.com/mono/monodevelop/pull/5614

**Pull request merged**